host-go/engine/tests/append_lens_test.go

```diff
@@ -49,7 +49,8 @@ func TestAppendLensWithoutWasm(t *testing.T) {
 				var arbitraryIndex module.MemSize = 5
 				return arbitraryIndex, nil
 			},
-			Transform: func(startIndex module.MemSize) (module.MemSize, error) {
+			Transform: func(next func() module.MemSize) (module.MemSize, error) {
```

Comment:
suggestion: I know you prefer straightforward/flat test bodies, but this one is very hard to read.

Looks like this `module.Instance` is just there to replicate some production code behaviour for testing purposes.
This is exactly what Fakes are for. Please consider extracting it into it's own struct. This will allow to reuse the struct/fake for other tests and will make it easier to read the test.

Reply:
I very much disagree here, this is testing public behaviour, and is also a nice way of documenting that this public behaviour is really painful for users.

Lens supports the writing of 'native' modules like this, and this test provides an example of what needs to be done in order to get it working, as any user would need to do so (as well as the other reasons for its existence noted in the description.

Masking how painful writing this is would devalue the test.